### PR TITLE
Ticket31705 merged

### PR DESCRIPTION
**Pull request by nmathewson**

### CODE REVIEW

File: scripts/coccinelle/test-operator-cleanup [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+# Copyright (c) 2001-2004, Roger Dingledine.
+# Copyright (c) 2004-2006, Roger Dingledine, Nick Mathewson.
+# Copyright (c) 2007-2019, The Tor Project, Inc.
+# See LICENSE for licensing information
+
+# This script looks for instances of C comparison operators as macro arguments,
+# and replaces them with our OP_* equivalents.
+#
+# Some macros that take operators are our tt_int_op() testing macro, and the
+# standard timercmp() macor.  Coccinelle can't handle their syntax, however,
```
Comment (teor2345) [inline scripts/coccinelle/test-operator-cleanup]:
```suggestion
# standard timercmp() macro.  Coccinelle can't handle their syntax, however,
```

File: src/app/config/config.c [hunk 2]
```diff
@@ -765,16 +765,20 @@ static const config_var_t option_vars_[] = {
...
 
 /** List of fallback directory authorities. The list is generated by opt-in of
  * relays that meet certain stability criteria.
  */
 static const char *default_fallbacks[] = {
+#ifndef COCCI
 #include "fallback_dirs.inc"
-  NULL
+#endif
+   NULL
```
Comment (teor2345) [inline src/app/config/config.c]: Spacing: why is this NULL indented 3 spaces?

File: src/app/config/config.c [hunk 3]
```diff
@@ -186,8 +186,12 @@ static const char unix_q_socket_prefix[] = "unix:\"";
 
 /** macro to help with the bulk rename of *DownloadSchedule to
  * *DowloadInitialDelay . */
+#ifndef COCCI
 #define DOWNLOAD_SCHEDULE(name) \
```
Comment (teor2345) [inline src/app/config/config.c]: The commit message doesn't talk about DOWNLOAD_SCHEDULE?

File: src/core/mainloop/periodic.h [hunk 4]
```diff
@@ -71,8 +71,10 @@ typedef struct periodic_event_item_t {
 } periodic_event_item_t;
 
 /** events will get their interval from first execution */
+#ifndef COCCI
```
Comment (teor2345) [inline src/core/mainloop/periodic.h]: Should this be in the periodic event commit?

File: src/lib/buf/buffers.c [hunk 5]
```diff
@@ -99,6 +99,7 @@
 #define DBG_S(s) (void)0
 #endif
 
+#ifndef COCCI
 #ifdef DISABLE_MEMORY_SENTINELS
```
Comment (teor2345) [inline src/lib/buf/buffers.c]: Should the other macro in the periodic event commit be in this commit instead?

File: scripts/coccinelle/tor-coccinelle.h [hunk 6]
```diff
@@ -57,4 +57,7 @@
 #define NS_DECL(a, b, c) a b c
 #define NS(a) a
 
+#define CONF_TEST_MEMBERS(a,b,c)
```
Comment (teor2345) [inline scripts/coccinelle/tor-coccinelle.h]: The commit message for this change is wrong: this is coccinelle, not coverity

File: scripts/coccinelle/test-operator-cleanup (hunk 1 above)
Comment (nmathewson) [reply scripts/coccinelle/test-operator-cleanup]: Fixup is e45fee1fefa6578375f2a83b8ba4ac66ea7c4e15

File: src/app/config/config.c (hunk 2 above)
Comment (nmathewson) [reply src/app/config/config.c]: 2df7442afd0d19377115f75d8a4b120092ba2a19 is the fixup.

**Pull request closed without merging**